atc/db/volume_repository.go

```diff
@@ -540,6 +542,53 @@ func (repository *volumeRepository) GetDestroyingVolumes(workerName string) ([]s
 	)
 }
 
+func (repository *volumeRepository) DestroyUnknownVolumes(workerName string, reportedHandles []string) (int, error) {
+	dbHandles, err := repository.queryVolumeHandles(sq.Eq{
+		"worker_name": workerName,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	unknownHandles := diff(reportedHandles, dbHandles)
+
+	if len(unknownHandles) == 0 {
+		return 0, nil
+	}
+
+	tx, err := repository.conn.Begin()
+	if err != nil {
+		return 0, err
+	}
+
+	defer Rollback(tx)
+
+	for _, unknownHandle := range unknownHandles {
+		_, err = psql.Insert("volumes").
+			Columns(
+				"handle",
+				"worker_name",
+				"state",
+			).
```

Comment:
Should we use an InsertBuilder outside the for loop similar to the pattern used for the container_repository.go